### ISSUE TITLE
Load predictor issue

### ISSUE THREAD

**Issue by Aklmenrah** [amaiya/ktrain#86]
Hi, 
I saved and downloaded my predictor from google colab, 
and then loaded it with 	
`loaded_predictor = ktrain.load_predictor('/tmp/schema_mapping')`

But then when I run the app with my Docker container i get this error:
`	AttributeError: 'NoneType' object has no attribute 'predict'
`

I also tried 
`loaded_predictor.model._make_predict_function()`

But then I get: 
`AttributeError: 'NoneType' object has no attribute 'model'
`

Any idea why I get that error? 
Thank you in advance.

**Comment by amaiya** [comment]
If you saved predictor with `predictor.save('/tmp/schema_mapping`), then there are two files saved':  `schema_mapping` and `schema_mapping.preproc` in your case.  Are you sure you've downloaded both?  If not, then `load_predictor` will print an error message and return `None` (which seems to be happening in your case).

(I have fixed this for next version so that an actual Exception is raised to avoid any confusion.)

**Comment by Aklmenrah** [comment]
I do have a folder tmp with in it schema_mapping.preproc and a folder schema_mapping (with config.json and tf_model.h5).
But it gives the same error. Am I placing the files in the wrong place?

**Comment by amaiya** [comment]
Yes, what you have is correct for Transformer models.  I just tried this on Google Colab.  It worked perfectly for me and I was not able to reproduce this issue.  

How are you downloading the predictor files?  
Are you using the same version of TensorFlow and ktrain locally as on Google Colab?

Here's what I did:

1. Train a model on Google Colab after manually installing TensorFlow 2.1.0 (`pip3 install tensorflow_gpu==2.1.0`) and ktrain (`pip3 install ktrain`):

```python
# load text data
categories = ['alt.atheism', 'soc.religion.christian','comp.graphics', 'sci.med']
from sklearn.datasets import fetch_20newsgroups
train_b = fetch_20newsgroups(subset='train', categories=categories, shuffle=True)
test_b = fetch_20newsgroups(subset='test',categories=categories, shuffle=True)
(x_train, y_train) = (train_b.data, train_b.target)
(x_test, y_test) = (test_b.data, test_b.target)

# build, train, and validate model (Transformer is wrapper around transformers library)
import ktrain
from ktrain import text
MODEL_NAME = 'distilbert-base-uncased'
t = text.Transformer(MODEL_NAME, maxlen=500, class_names=train_b.target_names)
trn = t.preprocess_train(x_train, y_train)
val = t.preprocess_test(x_test, y_test)
model = t.get_classifier()
learner = ktrain.get_learner(model, train_data=trn, val_data=val, batch_size=6)
learner.fit_onecycle(5e-5, 1)
```

2. Create and save a Predictor instance to a folder in my Google Drive:
```python
p = ktrain.get_predictor(learner.model, t)
p.save('/content/drive/My Drive/test_folder/mypred')
```
Note that you'll have to mount your Google Drive to Colab environment first.

3. Download predictor files (i.e., `mypred.preproc`, `mypred/config.json`, `mypred/ tf_model.h5`) and use the predictor on local machine:
```python
# cd to test_folder directory
import ktrain
p = ktrain.load_predictor('mypred')
p.predict('My computer monitor is blurry.')
# the above outputs 'comp.graphics'
```
The Predictor files were downloaded automatically, since I use an application to automatically sync Google Drive to local machine.

**Comment by Aklmenrah** [comment]
I am doing almost the same, and I do use same versions on colab and locally.

1. Train a model on colab

2. Save it and download it, even though for some reason colab raise an error when i try `files.download('/tmp/schema_mapping')`
But it's no big deal, I download the files from the directory on the left

3. I the save the files inside a folder where there is my app and everything for my Docker container.
-predict schema:
    -App,py
    -Docker stuff
    -tmp:
        -schema_mapping.preproc
        -schema_mapping:
            -confing.json
            -tf_model.h5

**Comment by amaiya** [comment]
Perhaps the `files.download` commanding is corrupting your file or something, which causes `transformers` library to return None for model.  I would check the `md5sum` of the model files on Google colab and compare with the ones you downloaded to ensure they are bit-for-bit the same.   

I did not use `files.download` to download the model files.  I used an app ([Backup and Sync](https://www.google.com/drive/download/backup-and-sync/)) that automatically syncs a Google Drive with a local folder.  So, on Google colab, I saved the model to my Google Drive, which, then, is automatically downloaded/synced to local folder.
